src/cmd/tools/split_index_shards/main/main.go

```diff
@@ -0,0 +1,233 @@
+// Copyright (c) 2021 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package main
+
+import (
+	"fmt"
+	iofs "io/fs"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/pborman/getopt"
+	"go.uber.org/zap"
+
+	"github.com/m3db/m3/src/dbnode/digest"
+	"github.com/m3db/m3/src/dbnode/generated/proto/index"
+	"github.com/m3db/m3/src/dbnode/persist/fs"
+	xerrors "github.com/m3db/m3/src/x/errors"
+	"github.com/m3db/m3/src/x/ident"
+	xos "github.com/m3db/m3/src/x/os"
+	xtime "github.com/m3db/m3/src/x/time"
+)
+
+var (
+	checkpointPattern = regexp.MustCompile(`/index/data/(\w+)/fileset-([0-9]+)-([0-9]+)-checkpoint\.db$`)
+	genericPattern    = regexp.MustCompile(`/index/data/(\w+)/fileset-([0-9]+)-([0-9]+)-.*\.db$`)
+)
+
+func main() {
+	var (
+		optPath       = getopt.StringLong("path", 'p', "", "Index path [e.g. /temp/lib/m3db/index]")
```

Comment:
nit: In other tools `--path` argument usually point one step higher (e.g. `/var/lib/m3db` and not `.../data`, `.../index`, etc. subdirectories). Maybe it would be worth to make this alike and append `/index` suffix when needed inside the code?

Reply:
Not appending `/index` suffix automatically allows us to specify index location more flexibly,
eg. `/temp/lib/m3db/index/data/aggregated_30d` would also work, if we need to process just a single namespace.